File: Backend/DAL/Entities/DeviceModel.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace DAL.Entities;

public record DeviceModel
{
    [Key] public string DeviceId { get; set; }
    public string Vendor { get; set; }
    public string Model { get; set; }
    public string OsVersion { get; set; }
    [JsonIgnore] public DateTime RegisteredAt { get; set; }
}

File: Frontend/Android/app/src/main/java/cz/stepes/smsgateway/models/DeviceModel.kt
package cz.stepes.smsgateway.models

import com.beust.klaxon.Json

data class DeviceModel(
    @Json(name = "deviceId") val deviceId: String,
    @Json(name = "vendor") val vendor: String,
    @Json(name = "model") val model: String,
    @Json(name = "osVersion") val osVersion: String
)